core/metadata.js

```diff
@@ -675,3 +678,21 @@ function shouldIgnore(
     isFolder: doc.docType === 'folder'
   })
 }
+
+function needsRevMigration(existing /*: Metadata */) /*: boolean %checks */ {
+  return existing.sides == null || existing.sides.rev == null
```

Comment:
Maybe explain that the first case is not supposed to happen but is here to make flow happy because of our permissive type annotations?

Reply:
I don't remember writing it for Flow's sake but I might have done it without thinking.
Are we sure this can never happen?